File: python/jzoffer-49.py
# -*- coding:utf-8 -*-
class Solution:
    def StrToInt(self, s):
        # write code here
        numlist=['0','1','2','3','4','5','6','7','8','9','+','-']
        label = 1
        res = 0
        if s == '' or s == '0':
            return 0
        for i in s:
            if i in numlist:
                if i == '+':
                    label = 1
                    continue
                elif i == '-':
                    label = -1
                    continue
                else:
                    res = res*10 + numlist.index(i)
            else:
                return 0
        return res*label

File: java/jzoffer-49.java
public class Solution {
    public int StrToInt(String str) {
        int n = str.length(), s = 1;
        int res = 0;
        if(n == 0) return 0;
        if(str.charAt(0) == '-') s = -1;
        for(int i = (str.charAt(0) ==  '-' || str.charAt(0) == '+') ? 1 : 0; i < n; ++i){
            if(!('0' <= str.charAt(i) && str.charAt(i) <= '9')) return 0;
            res=res*10+str.charAt(i)-'0';
        } 
        return res * s;
    }
}